test/with_dummyserver/test_socketlevel.py

```diff
@@ -152,6 +152,24 @@ def socket_handler(listener):
 
         self.assertEqual(http.pool.qsize(), http.pool.maxsize)
 
+    def test_read_timeout_dont_retry_method_not_in_whitelist(self):
+        timed_out = Event()
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+            while not sock.recv(65536):
```

Comment:
Yeah, a single `recv` is almost certainly fine: just enough to ensure the request doesn't get blocked behind the socket queue.